### PR TITLE
Update upgrade docs to discuss rolling and scheduled upgrades

### LINKED ISSUE #1919: [Request] Update the upgrade managed agent doc
## Description

Update the upgrade managed agent documentation, https://www.elastic.co/guide/en/fleet/current/upgrade-elastic-agent.html

Related to https://github.com/elastic/obs-dc-team/issues/739 

We made some changes to the upgrade feature:
* The feature is now GA 
* Upgrade could be scheduled 
* Upgrade will not be sent at all the agents at the same time but is now using a rolling window

## Collaboration

@nchaulet Can provide more info on the changes

### PR DESCRIPTION
**Pull request by dedemorton**
Closes #1919 

@nchaulet I have a few questions that I've added inline as comments. Can you respond? I'm on PTO next week, but if you can get this reviewed while I'm away, I'll get it merged the Monday I'm back. I tried to test but ran into a few snags with the binary download. Hopefully I got the steps right!

Preview of new topic: https://observability-docs_1938.docs-preview.app.elstc.co/guide/en/fleet/master/upgrade-elastic-agent.html

### CODE REVIEW

File: docs/en/ingest-management/fleet/upgrade-elastic-agent.asciidoc [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
-version.
+version. You can trigger upgrades to happen immediately, or specify a
+maintenance window during which the upgrades will occur. If your {stack}
+subscription level supports it, you can schedule upgrades to occur at a specific
+date and time.
+
+//REVIEWERS: From the design doc, it seemed like scheduled upgrades and
+//maintenance windows are only available to some subscription levels. But looking
+//at a snapshot build, seemed like that might not be true. Can you clarify?
+//If the upgrade procedure is different for Platinum than other users, I guess
```
Comment (nchaulet) [inline docs/en/ingest-management/fleet/upgrade-elastic-agent.asciidoc:20]: Yes only platinium user can schedule upgrade, we miss this during implementation but it should be fixed by https://github.com/elastic/kibana/pull/134833

File: docs/en/ingest-management/fleet/upgrade-elastic-agent.asciidoc [hunk 2]
```diff
@@ -1,15 +1,25 @@
...
 With {fleet} upgrade capabilities, you can view and select agents that are out
 of date, and trigger selected agents to download, install, and run the new
-version.
+version. You can trigger upgrades to happen immediately, or specify a
+maintenance window during which the upgrades will occur. If your {stack}
+subscription level supports it, you can schedule upgrades to occur at a specific
+date and time.
+
+//REVIEWERS: From the design doc, it seemed like scheduled upgrades and
+//maintenance windows are only available to some subscription levels. But looking
```
Comment (nchaulet) [inline docs/en/ingest-management/fleet/upgrade-elastic-agent.asciidoc:18]: Maintenance windows are available at all subscription level (only the schedule part is licence restricted)

File: docs/en/ingest-management/fleet/upgrade-elastic-agent.asciidoc [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
+//at a snapshot build, seemed like that might not be true. Can you clarify?
+//If the upgrade procedure is different for Platinum than other users, I guess
+//I can break this into two sections: "Upgrade to latest version" and something like
+//"Rolling upgrades"
```
Comment (dedemorton) [inline docs/en/ingest-management/fleet/upgrade-elastic-agent.asciidoc:23]:
```suggestion
```

**Comment by dedemorton** [comment]
@nchaulet Have you reviewed the content?

**Review by nchaulet — APPROVED** [review]
LGTM 🚀

**Pull request merged**